### ISSUE TITLE
TypeError when NEP18 is enabled

### ISSUE THREAD

**Issue by liwt31** [cupy/cupy#2130]
When calling type related functions borrowed from numpy such as `cp.result_type` or `cp.can_cast` and NEP18 is enabled, a type error will be raised.

* Conditions
```
CuPy Version          : 6.0.0b3
CUDA Root             : /usr/local/cuda-10.0
CUDA Build Version    : 10000
CUDA Driver Version   : 10010
CUDA Runtime Version  : 10000
cuDNN Build Version   : None
cuDNN Version         : None
NCCL Build Version    : None
NCCL Runtime Version  : None
```
* Code to reproduce
```
import os
os.environ["NUMPY_EXPERIMENTAL_ARRAY_FUNCTION"] = "1"
import cupy as cp
a = cp.random.rand(10)
print(cp.result_type(a,a))
```
* Error messages, stack traces, or logs
```
$ python test.py
Traceback (most recent call last):
  File "test.py", line 5, in <module>
    print(cp.result_type(a,a))
  File "/home/wtli/.conda/envs/cp/lib/python3.6/site-packages/numpy/core/overrides.py", line 165, in public_api
    implementation, public_api, relevant_args, args, kwargs)
TypeError: no implementation found for 'numpy.result_type' on types that implement __array_function__: [<class 'cupy.core.core.ndarray'>]
```
* Further info
The following lines lead to the error:
https://github.com/cupy/cupy/blob/6ee8db22af9e9ba41b84a3e16fc5ebde1c3c4ff8/cupy/core/core.pyx#L1257-L1259
As functions like `result_type` are borrowed from numpy directly, this will return a NotImplemented, and thus a error. These lines are introduced in #2024.

**Comment by liwt31** [comment]
I've read through the discussions in #2029 and it seems that an easy solution isn't coming any time soon. I'll simply leave the issue here for future reference.

**Comment by kmaehashi** [comment]
Currently NumPy team is working on fixing this issue: https://github.com/cupy/cupy/issues/2029
We will continue to check how it goes.

**Comment by takagi** [comment]
This PR can be closed as #2249 solved it.
